core/src/main/kotlin/api/OrganizationsRoute.kt

```diff
@@ -319,5 +325,65 @@ fun Route.organizations() = route("organizations") {
                 }
             }
         }
+
+        route("groups") {
```

Comment:
There are different ways to structure these endpoints, and I am not sure which is the best one from a usability PoV. You are approaching it from the groups. One alternative could be having a /users endpoint. In order to add a user with a specific permission, one would send a PUT to this endpoint with a body containing the user ID and the permission. One advantage I would see here is that a GET on users would directly return all relevant information about users.

Reply:
I got your point. When I think of the user management aspect from the UI viewpoint on the other hand, in future I might want to have a GET operation on _organizations/{organizationId}/groups/..._ level to get a list of users that have some kind of group membership there. So, maybe in future something like _GET organizations/{organizationId}/groups/readers_ returning a list of users in the _readers_ group on organization level.

Reply:
Unfortunately, I am no expert for UX. Probably both approaches have their pros and cons, and there are different use cases that can be handled better with either of them.

It is certainly a valid use case to list all users in a specific group. It might also be useful to search for a specific user in the whole set of users. Would it be perhaps easier for the UI to have one endpoint that returns all information and then do the filtering for different presentations locally?

Reply:
I would say, not "search for a specific user in the whole set of users" , but "search for a user in all the users of an Organization", because we have multi-tenancy and there are rare (system admin) cases where all users shall be shown.